scripts/generate-clients/copy-to-clients.js

```diff
@@ -25,6 +32,28 @@ const getOverwritablePredicate = packageName => pathName => {
   );
 };
 
+const mergeManifest = (fromContent, toContent) => {
+  const merged = {};
+  const toNames = Object.keys(toContent);
+  for (const name of toNames) {
+    if (!fromContent[name]) {
+      merged[name] = toContent[name];
+      continue;
+    }
+    if (typeof toContent[name] === "object") {
+      merged[name] = mergeManifest(fromContent[name], toContent[name]);
+    } else {
+      merged[name] = toContent[name] || fromContent[name];
```

Comment:
Add comment, say:
```javascript
  // If key (say dependency) is present in both codegen and package.json, we prefer latter
```

Reply:
Added in https://github.com/aws/aws-sdk-js-v3/pull/900/commits/2ae427a4373225ca7637c146a568f5136b4473ea